xiaogpt/xiaogpt.py

```diff
@@ -132,10 +132,13 @@ async def _init_data_hardware(self):
         # to set device id
 
         for h in hardware_data:
-            if did := os.getenv("MI_DID"):
+            did = os.getenv("MI_DID") or (self.config.mi_did if self.config.mi_did != '' else None)
+            if did:
```

Comment:
```suggestion
            if did := self.config.mi_did:
```
The env var is already read by the config, you don't need to look at it again.